### PR TITLE
feat(ClearAll): add withQuery to also clear the search query

### PR DESCRIPTION
**Pull request by mthuret**
https://deploy-preview-1958--algolia-instantsearch.netlify.com/instantsearch.js/react/storybook/?selectedKind=ClearAll&selectedStory=clear%20all%20refinements%20and%20the%20query&full=0&down=1&left=1&panelRight=1&downPanel=kadirahq%2Fstorybook-addon-knobs

### CODE REVIEW

File: packages/react-instantsearch/src/connectors/connectCurrentRefinements.js [hunk 1]
```diff
@@ -21,16 +23,19 @@ export default createConnector({
     const items = metadata.reduce((res, meta) =>
         typeof meta.items !== 'undefined' ? res.concat(meta.items) : res
       , []);
+    const query = props.withQuery && searchResults.results ? searchResults.results.query : undefined;
```
Comment (vvo) [inline packages/react-instantsearch/src/connectors/connectCurrentRefinements.js]: naming proposal alternative: `clearsQuery`

```jsx
<ClearAll withQuery />
// vs
<ClearAll clearsQuery />
```

Comment (mthuret) [inline packages/react-instantsearch/src/connectors/connectCurrentRefinements.js]: Yes, oki.

**Comment by vvo** [comment]
LGTM

**Comment by Kikobeats** [comment]
Oh nice! I'm going to update my code this week, feedback soon! Thanks :-)

**Pull request merged**